app/src/main/java/de/triology/cas/ldap/CesAuthenticationEventExecutionPlanConfiguration.java

```diff
@@ -0,0 +1,226 @@
+        }
+
+        val passwordPolicy = l.getPasswordPolicy();
+        if (passwordPolicy.isEnabled()) {
+            LOGGER.trace("Password policy is enabled for [{}]. Constructing password policy configuration", l.getLdapUrl());
+            val cfg = createLdapPasswordPolicyConfiguration(passwordPolicy, authenticator, multiMapAttributes);
+            handler.setPasswordPolicyConfiguration(cfg);
+        }
+
+        val attributes = CollectionUtils.wrap(multiMapAttributes);
+        handler.setPrincipalAttributeMap(attributes);
+
+        LOGGER.debug("Initializing LDAP authentication handler for [{}]", l.getLdapUrl());
+        handler.initialize();
+
+        return handler;
+    }
+
+    @Override
+    public void configureAuthenticationExecutionPlan(AuthenticationEventExecutionPlan plan) {
+        plan.registerAuthenticationHandler(cesGroupAwareLdapAuthenticationHandler());
+        LOGGER.trace("Registered {}, registered authentication handlers: {}",
+                CesGroupAwareLdapAuthenticationHandler.class.getSimpleName(), plan.getAuthenticationHandlers());
+    }
+
+    // 1:1 copy from LdapAuthenticationConfiguration
+    private static PasswordPolicyContext createLdapPasswordPolicyConfiguration(final LdapPasswordPolicyProperties passwordPolicy,
+                                                                               final Authenticator authenticator,
+                                                                               final Multimap<String, Object> attributes) {
+        val cfg = new PasswordPolicyContext(passwordPolicy);
+        val handlers = new HashSet<>();
+
+        val customPolicyClass = passwordPolicy.getCustomPolicyClass();
+        if (StringUtils.isNotBlank(customPolicyClass)) {
+            try {
+                LOGGER.debug("Configuration indicates use of a custom password policy handler [{}]", customPolicyClass);
+                val clazz = (Class<AuthenticationResponseHandler>) Class.forName(customPolicyClass);
+                handlers.add(clazz.getDeclaredConstructor().newInstance());
+            } catch (final Exception e) {
+                LoggingUtils.warn(LOGGER, "Unable to construct an instance of the password policy handler", e);
+            }
+        }
+        LOGGER.debug("Password policy authentication response handler is set to accommodate directory type: [{}]", passwordPolicy.getType());
+        switch (passwordPolicy.getType()) {
+            case AD:
+                handlers.add(new ActiveDirectoryAuthenticationResponseHandler(Period.ofDays(cfg.getPasswordWarningNumberOfDays())));
+                Arrays.stream(ActiveDirectoryAuthenticationResponseHandler.ATTRIBUTES).forEach(a -> {
+                    LOGGER.debug("Configuring authentication to retrieve password policy attribute [{}]", a);
+                    attributes.put(a, a);
+                });
+                break;
```

Comment:
This method is quite long but when I look at this strategy-like pattern here, it looks very easy to extract the single password handler routines into their own methods. That could introduce a better readability of this method.

Reply:
In principle, I agree with you. 

However, this code is an exact 1:1 copy of the CAS. I only copied this code because it was not possible to call this method because the method is private. This method should actually be public in CAS. There should simply be no changes here. 
This is not a great solution, but there is probably no better one at the moment.

Therefore, I leave this method unchanged.